### PR TITLE
Add attribute "algoliaLastUpdateCET" to all objects indexed via the e…

### PR DESCRIPTION
**Pull request by JanPetr**
…xtension. It contains the exact CET when the record was update for the last time. It helps to debug indexing issues.

Implements #729

### CODE REVIEW

**Comment by maxiloc** [comment]
I would actually do it here https://github.com/algolia/algoliasearch-magento/blob/master/app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php#L155-L163

**Comment by JanPetr** [comment]
@maxiloc I was thinking about doing it there, but I didn't what to iterate again over all the objects.

File: app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php [hunk 1]
```diff
@@ -109,52 +109,37 @@ public function mergeSettings($index_name, $settings)
...
+    public function addObjects($objects, $indexName)
     {
-        $long_attributes = array('description', 'short_description', 'meta_description', 'content');
+        $currentCET = new DateTime('now', new DateTimeZone('Europe/Paris'));
+        $currentCET = $currentCET->format('Y-m-d H:i:s');
+        
+        $modifiedIds = array();
 
-        $good_size = true;
+        foreach ($objects as $key => $object) {
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php]: I would do this loop in another function

Comment (JanPetr) [inline app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php]: Why?
The reason the loop is here I don't want to mix "add time" and "handle big records" features together in any other function. The only thing those "features" have in common is the "addObjects" method.

Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php]: To keep it very simple like other method of the file:
https://github.com/algolia/algoliasearch-magento/blob/221492f938275674b3b0f4865a26dc1f8ece2ffd/app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php#L34-L76
that are just a bridge where we can do stuff

I tend to prefer seeing:
```
  if ($this->config->isPartialUpdateEnabled()) {
            $index->partialUpdateObjects($objects);
        } else {
            $index->addObjects($objects);
        }
```
very quickly and not at the end of a long function

The method name can be generic like `alterRecords` or `modifyRecords` and inside you can have two different submethods

Comment (JanPetr) [inline app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php]: Makes sense, stay tuned

Comment (JanPetr) [inline app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php]: @maxiloc Extracted, can you review?

File: app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php [hunk 2]
```diff
@@ -109,52 +109,37 @@ public function mergeSettings($index_name, $settings)
...
         }
 
-        if ($good_size === false) {
+        if (!empty($modifiedIds)) {
             /** @var Mage_Adminhtml_Model_Session $session */
             $session = Mage::getSingleton('adminhtml/session');
-            $session->addError('Algolia reindexing : You have some records ('.implode(',',
-                        array_keys($ids)).') that are too big. They have either been truncated or skipped');
+            $session->addError('Algolia reindexing : You have some records ('.implode(',', $modifiedIds).') that are too big. They have either been truncated or skipped');
         }
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php]: I think you are missing the return

**Pull request merged**